### DISCUSSION TITLE
Error: P1001: Can't reach database server at `aws.connect.psdb.cloud`:`3306`

### DISCUSSION THREAD

**Discussion by sidharthhhh** [planetscale/discussion#501] [FAQs]
i created a new project in planet scale and paste the url in .env
when i use this command "npx prisma db push" then it is throughing error "Error: P1001: Can't reach database server at `aws.connect.psdb.cloud`:`3306`

Please make sure your database server is running at `aws.connect.psdb.cloud`:`3306`"

how can i resolve this error?!

**Comment by jgreet** [comment]
The first thing to try is increasing the connection timeout from its default of 5 seconds to something like 30. You can do that by adding a `connect_timeout` parameter to your connection string.

```
mysql://USER:PASSWORD@HOST:PORT/DATABASE?connect_timeout=30
```


See also https://planetscale.com/docs/prisma/prisma-best-practices#prisma-error-p1001

**Comment by bijaydas** [comment]
Hi, @sidharthhhh  I am facing a similar error. I am trying to migrate using Prisma 5.0. But I am getting this error. Can you please share how you solved it?
![Screenshot from 2023-07-29 01-34-07](https://github.com/planetscale/discussion/assets/20036746/892d61f6-ba86-4a5c-a682-9efdf8f528c3)

**Reply by sidharthhhh** [reply]
Just change the password, it reset in every 24 hr...

**Reply by bijaydas** [reply]
[solved] fixed it using a shadow database in Prisma

**Reply by capt-amlan-padhi** [reply]
@bijaydas how did you do it? can you please tell me?

**Reply by bijaydas** [reply]
I have created a another database on PlanetScale (You can create anywhere, even in your local machine). And use that database as Shadow Database in your `.env` with `SHADOW_DATABASE_URL`.

You can learn more about it here [Prisma Docs](https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database).

**Reply by capt-amlan-padhi** [reply]
@bijaydas I am having a different problem
lemme show you my code:

.env:
```DATABASE_URL='mysql://<username>:<password>@aws.connect.psdb.cloud:3306/blog-ai-app?connect_timeout=30&&ssl={"rejectUnauthorized":true}&&sslcert=/etc/ssl/cert.pem'```

schema.prisma:
```


generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model Post {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String
  category  String
  content   String   @db.Text
  author    String
  image     String
  snippet   String   @db.Text
}
```

![Screenshot 2023-08-22 001141](https://github.com/planetscale/discussion/assets/94183639/6639d5cf-e1f0-44cc-92e8-0c6d4d8cd493)

**Reply by bijaydas** [reply]
Here are the possible solutions you can try

1. If you don't have pscale cli installed, please do. it will be easier to access.
2. After installing the `cli` try to login `pscale shell your-database-name branch-name`
3. Check if you can login via `mysql -u your-user-name -p your-password`
4. Check if you can query your database in web console.

Let me know the update
